### DISCUSSION TITLE
Y up or down ?

### DISCUSSION THREAD

**Discussion by nyue** [cduck/drawSvg#66] [Q&A] [answered]
I have the following code, depending on whether I concatenate the commands or call append separately, the Y direction is opposite, have my misunderstood the API ?
```
import drawSvg as draw

d = draw.Drawing(400, 400, origin='center', displayInline=False)

p1 = draw.Path(stroke='green', stroke_width=2, fill='transparent')
separate = False
if separate:
    p1.append('m', 31.75, 42.244047)
    p1.append('c', 0, 0, 68.055716, 12.241243, 97.51786, 35.529762)
else:
    p1.m(31.75, 42.244047).c(0, 0, 68.055716, 12.241243, 97.51786, 35.529762)
d.append(p1)

d.saveSvg('myspline.svg')
d.savePng('myspline.png')
```

**Comment by cduck** [comment, marked as answer]
drawSvg uses an upward pointing y-axis (see #11).  You should use the named methods `Path.m()`, `Path.c()`, etc.  `Path.append()` is a lower-level API that you should avoid.

**Reply by nyue** [reply]
If I am programmatically reading in data, what is a good way to provide data via the drawSVG API to Path object ?

I won't know in advanced how many pieces of information.

I was looking at calling append() since I am iterating through a set of data.

Cheers

**Reply by cduck** [reply]
OK.  You can use something like `getattr(mypath, 'm')(31.75, 42.244047)` to indirectly call the method.  The benefit of this is it will validate the data you pass (i.e. throw an error if you give the wrong number of arguments for a command).

**Reply by nyue** [reply]
Got it working. Both version looks the same now

```
import drawSvg as draw

def main(separate: bool):
    d = draw.Drawing(400, 400, origin='center', displayInline=False)

    p = draw.Path(stroke='green', stroke_width=2, fill='transparent')
    if separate:
        getattr(p,'m')(31.75, 42.244047)
        getattr(p,'c')(0, 0, 68.055716, 12.241243, 97.51786, 35.529762)
    else:
        p.m(31.75, 42.244047).c(0, 0, 68.055716, 12.241243, 97.51786, 35.529762)
    d.append(p)

    d.saveSvg('myspline_{}.svg'.format(separate))
    d.savePng('myspline_{}.png'.format(separate))

if __name__ == "__main__":
    # execute only if run as a script
    main(True)
    main(False)
```
